File: runtime/botmanager/src/com/newbound/code/primitive/object/Get.java
package com.newbound.code.primitive.object;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import com.newbound.code.primitive.Primitive;

public class Get extends Primitive 
{
	public Get() throws JSONException {
		super("{ in: { a: {}, b: {} }, out: { a: {} } }");
	}
	
	@Override
	public JSONObject execute(JSONObject in) 
	{
		JSONObject out = new JSONObject();
		try
		{
			Object a = in.get("a");
			if (a instanceof JSONArray)
			{
				int b = in.getInt("b");
				Object c = ((JSONArray)a).get(b);
				out.put("a", c);
			}
			else
			{
				String b = in.getString("b");
				Object c = ((JSONObject)a).get(b);
				out.put("a", c);
			}
		}
		catch (Exception x) { x.printStackTrace(); }

		return out;
	}
}

File: runtime/botmanager/src/newbound/code/primitive/object/get.py
from ..primitive import Primitive


class Get(Primitive):

    def __init__(self):
        super().__init__('{"in": {"a": {}, "b", {}}, "out": { "c": {} } }')

    def execute(self, input):
        out = {}
        a = input.get("a")
        if isinstance(a, list):
            b = input["b"]
            c = a[b]
        else:
            # dict assumed
            b = input["b"]
            c = a[b]
        out["c"] = c
        return out
